jetty-util/src/main/java/org/eclipse/jetty/util/Pool.java

```diff
@@ -215,6 +224,42 @@ public Entry acquire()
         return null;
     }
 
+    /**
+     * Utility method to acquire an entry from the pool,
+     * reserving and creating a new entry if necessary.
+     *
+     * @param creator a function to create the pooled value for a reserved entry.
+     * @return an entry from the pool or null if none is available.
+     */
+    public Entry acquire(Function<Entry, T> creator)
+    {
+        Entry entry = acquire();
+        if (entry != null)
+            return entry;
+
+        Reservation reservation = reserve(getMaxEntries());
+        if (reservation == null)
+            return null;
+
+        T value = null;
+        try
+        {
+            value = creator.apply(reservation.getEntry());
+        }
+        catch (Throwable th)
+        {
+            LOGGER.warn(th);
+        }
+
+        if (value == null)
+        {
+            reservation.remove();
+            return null;
+        }
+
+        return reservation.acquire(value);
```

Comment:
It may be possible that the pool is closed just before the call to this method, so we should return `null`, which means `reservation.acquire()` should return a `boolean`.

Reply:
Currently it throws an ISE, which is what enable did before.  reservation.acquire can't return a boolean because it returns the entry.

Reply:
But the acquire will throw ISE if the pool has been closed... just as it did before.